tabbycat/results/result.py

```diff
@@ -519,57 +438,45 @@ def save(self):
             da = self.debateadjs[adj]
             for side in self.sides:
                 dt = self.debateteams[side]
-                for pos in self.positions:
-                    self.ballotsub.speakerscorebyadj_set.update_or_create(
-                        debate_team=dt, debate_adjudicator=da, position=pos,
-                        defaults=dict(score=self.get_score(adj, side, pos)))
+                self.ballotsub.teamscorebyadj_set.update_or_create(
+                    debate_team=dt, debate_adjudicator=da,
+                    defaults=self.get_defaults_fields('teamscorebyadj', adj, side))
 
     # --------------------------------------------------------------------------
     # Data setting and retrieval
     # --------------------------------------------------------------------------
 
-    def get_score(self, adjudicator, side, position):
-        try:
-            return self.scoresheets[adjudicator].get_score(side, position)
-        except AttributeError:
-            logger.exception("Tried to get score, but scoresheet doesn't do scores. "
-                    "self.takes_scores is %s", self.takes_scores)
-            return None
-
-    def set_score(self, adjudicator, side, position, score):
-        try:
-            scoresheet = self.scoresheets[adjudicator]
-        except KeyError:
-            logger.exception("Tried to set score by adjudicator %s, but this adjudicator "
-                "doesn't have a scoresheet.", adjudicator)
-            return
-        try:
-            scoresheet.set_score(side, position, score)
-        except AttributeError:
-            logger.exception("Tried to set score, but scoresheet doesn't do scores. "
-                    "self.takes_scores is %s", self.takes_scores)
+    def get_winner(self, adjudicator):
+        return next(iter(self.scoresheets[adjudicator].winners()), None)
```

Comment:
Is there an assumption here that there is at most one winner whenever this function is called? If so, it's probably better to include an `assert` statement that verifies that `.winners()` has at most one item, and then return that item using `.pop()`, rather than using `next(iter(...))`. That way if the condition fails, it fails fast, and also it'll be more explicit to read.

Reply:
No, there may be either 0 or 1 winners, and if there are no winners, `None` should be returned. `.pop()` couldn't be used as it would mutate the set.

Reply:
I said _at most_ one winner 😉 as in, there can't be more than one, though there might be less than one. But good point re `pop()`—kinda dissatisfying that `next(iter(...))` is the cleanest way otherwise, but it is what it is! I still think you want to `assert` assumptions to get it to fail fast though, or at least get the logger to throw an error at Sentry (if you think crashing due to the state is worse than ploughing ahead silently).